### PR TITLE
token auth, login and logout

### PR DESCRIPTION
**Pull request by ps-eqsdigital**
token authentication
      middleware for jwt authorize
login 
      returns a jwt token
logout
      sets user token to invalid

### CODE REVIEW

File: api/Business/BusinessModels/UserBusinessModel.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        public UserBusinessModel() { }
+
+        public UserBusinessModel(User user)
+        {
+            Uuid = user.Uuid;
+            Name=user.Name;
+            Email=user.Email;
+            Picture=user.Picture;
+        }
+
```
Comment (tentone) [inline api/Business/BusinessModels/UserBusinessModel.cs:25]: Add docs to models

File: api/Business/BusinessObjects/UserBusinessObject.cs [hunk 2]
```diff
@@ -86,6 +110,66 @@ public async Task<OperationResult<CreateUserBusinessModel>> Insert(User record)
...
+                        IsValid = true
+                    };
+                    await _genericDataAccessObject.UpdateAsync<UserTokenAuthentication>(userToken);
+                }
+                else
+                {
+                    userToken.Token = token;
+                    userToken.IsValid = true;
+                    await _genericDataAccessObject.UpdateAsync<UserTokenAuthentication>(userToken);
+                }
```
Comment (tentone) [inline api/Business/BusinessObjects/UserBusinessObject.cs]:
```suggestion

                if(userToken == null)
                {
                    userToken = new UserTokenAuthentication
                    {
                        UserId = result.Id,
                        Token = token,
                        IsValid = true
                    };
                }
                else
                {
                    userToken.Token = token;
                    userToken.IsValid = true;
                   
                }
                 await _genericDataAccessObject.UpdateAsync<UserTokenAuthentication>(userToken);
```

**Review by tentone — CHANGES_REQUESTED** [review]

File: api/Business/BusinessObjects/UserBusinessObject.cs (hunk 2 above)
Comment (ps-eqsdigital) [reply api/Business/BusinessObjects/UserBusinessObject.cs]: fixed

File: api/Business/BusinessObjects/UserBusinessObject.cs [hunk 3]
```diff
@@ -56,15 +71,17 @@ public async Task<OperationResult> Update(Guid uuid, User record)
...
+        public async Task<OperationResult<List<UserBusinessModel>>> ListFilteredUsers(string search, int sort)
         {
             return await ExecuteOperation(async () => {
 
                 if (sort !=0 && sort != 1)
                 {
                     throw new Exception();
                 }
-                List<User> result =await _userDataAccessObject.FilterUsers(search,sort);
+                List<User> users =await _userDataAccessObject.FilterUsers(search,sort);
```
Comment (js-eqsdigital) [inline api/Business/BusinessObjects/UserBusinessObject.cs:84]: space

File: api/Business/BusinessObjects/UserBusinessObject.cs [hunk 4]
```diff
@@ -86,6 +110,66 @@ public async Task<OperationResult<CreateUserBusinessModel>> Insert(User record)
...
+                {
+                    throw new Exception("Invalid email");
+                }
+
+                User result = await _userDataAccessObject.GetUserByEmail(email);
+                if (result == null)
+                {
+                    throw new Exception("No email found");
+                }
+                if (result.Password != password)
```
Comment (js-eqsdigital) [inline api/Business/BusinessObjects/UserBusinessObject.cs]: password should be encrypted in db

**Review by js-eqsdigital — CHANGES_REQUESTED** [review]

File: api/Business/BusinessObjects/UserBusinessObject.cs (hunk 4 above)
Comment (jocko10) [reply api/Business/BusinessObjects/UserBusinessObject.cs]: fixed

File: api/Business/BusinessObjects/UserBusinessObject.cs (hunk 3 above)
Comment (jocko10) [reply api/Business/BusinessObjects/UserBusinessObject.cs:84]: fixed

**Review by js-eqsdigital — APPROVED** [review]

**Pull request merged**